File: Moongy.RD.Launchpad.Data/Entities/CharacteristicInContractVariant.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Moongy.RD.Launchpad.Data.Base;

namespace Moongy.RD.Launchpad.Data.Entities;

public class CharacteristicInContractVariant : Entity
{

    [ForeignKey(nameof(ContractVariant))]
    public int ContractVariantId { get; set; }

    [ForeignKey(nameof(ContractCharacteristic))]
    public int ContractCharacteristicId { get; set; }

    public virtual ContractVariant? ContractVariant { get; set; }

    public virtual ContractCharacteristic? ContractCharacteristic { get; set; }
    public string Value { get; set; }
}

File: app/src/models/CharacteristicInContractVariant.ts
import { ContractCharacteristic } from "./ContractCharacteristic";
import { ContractVariant } from "./ContractVariant";
import { Entity } from "./Entity";

export type CharacteristicInContractVariant = Entity & {
    contractCharacteristic: ContractCharacteristic,
    contractVariant: ContractVariant,
    Value: string
}